### ISSUE TITLE
panic: value is null  while trying local-exec

### ISSUE THREAD

**Issue by paroque28** [hashicorp/terraform#28575]
### Terraform Version

```
2021-05-01T00:56:24.811Z [INFO]  Terraform version: 0.15.0 dev
2021-05-01T00:56:24.811Z [INFO]  Go runtime version: go1.13
```

### Terraform Configuration Files

```terraform
resource "null_resource" "x" {
  depends_on = [
    azurerm_container_registry.acr,
  ]
  for_each = local.manifest.images
  triggers = {
    always_run = timestamp()
  }
  provisioner "local-exec" {
    command = "scripts/clone_repos.sh"
    environment = {
      X_ECR = local.x_registry_url
      X_REGION = local.x_region
      DEST_REGISTRY = local.local_registry_url
      DEST_REGISTRY_USERNAME = local.local_registry_username
      DEST_REGISTRY_PASSWORD = local.local_registry_password
      SOURCE = format("%s/%s", local.x_project, each.key)
      DEST =  each.key
      TAGS = join(" ", each.value)
    }
  }
}
```

### Crash Output

````
2021-05-01T00:57:10.956Z [TRACE] applyProvisioners: provisioning null_resource.x["w_installer"] with "local-exec"
2021-05-01T00:57:10.956Z [TRACE] vertex "null_resource.x[\"digital-twin/12541\"]": visit complete

panic: value is null

goroutine 304 [running]:
github.com/zclconf/go-cty/cty.Value.AsString(0x2fcb820, 0xc0000e6390, 0x0, 0x0, 0x1, 0xc000aa73a0)
        /go/pkg/mod/github.com/zclconf/go-cty@v1.8.0/cty/value_ops.go:1254 +0x16e
github.com/hashicorp/terraform/builtin/provisioners/local-exec.(*provisioner).ProvisionResource(0xc00032b140, 0x2fcb920, 0xc000aa7290, 0x24d5f00, 0xc001616360, 0x2fcb920, 0xc000aa7080, 0x0, 0x0, 0x2f5d880, ...)
        /go/terraform-main/builtin/provisioners/local-exec/resource_provisioner.go:85 +0x1219
github.com/hashicorp/terraform/terraform.(*NodeAbstractResourceInstance).applyProvisioners(0xc000c128c0, 0x30236c0, 0xc0010185b0, 0xc00067ae40, 0x1, 0xc0000ebcc0, 0x1, 0x1, 0x0, 0x0)
        /go/terraform-main/terraform/node_resource_abstract_instance.go:1732 +0x862
github.com/hashicorp/terraform/terraform.(*NodeAbstractResourceInstance).evalApplyProvisioners(0xc000c128c0, 0x30236c0, 0xc0010185b0, 0xc00067ae40, 0xc000ba3c01, 0x1, 0x2, 0x0, 0x0)
        /go/terraform-main/terraform/node_resource_abstract_instance.go:1583 +0x20a
github.com/hashicorp/terraform/terraform.(*NodeApplyableResourceInstance).managedResourceExecute(0xc00087a480, 0x30236c0, 0xc0010185b0, 0x7a, 0x0, 0x0)
        /go/terraform-main/terraform/node_resource_apply_instance.go:286 +0x1109
github.com/hashicorp/terraform/terraform.(*NodeApplyableResourceInstance).Execute(0xc00087a480, 0x30236c0, 0xc0010185b0, 0x7efce4080401, 0x8, 0xc00128aba0, 0xbc5954)
        /go/terraform-main/terraform/node_resource_apply_instance.go:129 +0xc6
github.com/hashicorp/terraform/terraform.(*ContextGraphWalker).Execute(0xc00052d380, 0x30236c0, 0xc0010185b0, 0x7efce4080488, 0xc00087a480, 0x0, 0x0, 0x0)
        /go/terraform-main/terraform/graph_walk_context.go:127 +0xde
github.com/hashicorp/terraform/terraform.(*Graph).walk.func1(0x28fa740, 0xc00087a480, 0x0, 0x0, 0x0)
        /go/terraform-main/terraform/graph.go:59 +0xa0a
github.com/hashicorp/terraform/dag.(*Walker).walkVertex(0xc000a4c000, 0x28fa740, 0xc00087a480, 0xc000a1c940)
        /go/terraform-main/dag/walk.go:381 +0x2e2
created by github.com/hashicorp/terraform/dag.(*Walker).Update
        /go/terraform-main/dag/walk.go:304 +0x11ef
````

### Steps to Reproduce

1. `terraform init`
2. `terraform apply`

**Comment by paroque28** [comment]
Program is failing here: 
https://github.com/hashicorp/terraform/blob/v0.15.0/builtin/provisioners/local-exec/resource_provisioner.go#L85
and here:
https://github.com/zclconf/go-cty/blob/v1.8.0/cty/value_ops.go#L1254

What could it be?

**Comment by paroque28** [comment]
Is it just that one of my env variables inside the environment is null?
If so, this could be handled as an exception.

**Comment by paroque28** [comment]
Yes, that was it.

local.x_registry_url was equal to null

**Comment by jbardin** [comment]
Hi @paroque28,

Fixed in the latest release by #28457
